core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -993,28 +1019,40 @@ public void updateBufferSize(ByteSizeValue shardIndexingBufferSize, ByteSizeValu
         if (preValue.bytes() != shardIndexingBufferSize.bytes()) {
             // so we push changes these changes down to IndexWriter:
             engine.onSettingsChanged();
+            logger.debug("updating index_buffer_size from [{}] to [{}]", preValue, shardIndexingBufferSize);
```

Comment:
> can we make this log include the decision about whether to refresh (and also the current IW memory consumption, while at it)

You mean have a single debug log line instead of two?  OK ...